Migrate library to PyLoadAPI 1.1.0 in pyLoad integration

tr4nt0r opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Replaces the API Library, which is currently embedded into `sensor.py`, with an external async library. 

This is part of a somewhat bigger refactoring of the integration (see https://github.com/home-assistant/core/pull/115260), but I'm splitting it up into smaller chunks. This PR introduces a new library and otherwise only parts of the code got improved, that i had to touch anyway and left the rest as is.  Next PR will then introduce configuration via UI (config flow).

External API:

https://github.com/tr4nt0r/PyLoadAPI/releases/tag/1.1.0


Had to resubmit a new PR as I messed up a rebase in the previous PR  (#115652) and too many people got subscribed to the PR

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.

  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Commit:
Migrate pyLoad integration to externa API library

Commit:
Add const to .coveragerc

Commit:
raise update failed when cookie expired

Commit:
fix exceptions

Commit:
Add tests

Commit:
bump to PyLoadAPI 1.1.0

Commit:
remove unreachable code

Commit:
fix tests

tr4nt0r commented:
@coderabbitai review

joostlek commented on homeassistant/components/pyload/manifest.json:
```diff
@@ -1,7 +1,10 @@
-  "iot_class": "local_polling"
+  "integration_type": "service",
+  "iot_class": "local_polling",
+  "loggers": ["pyloadapi.api"],
```
Haven't ever seen a logger with a period in it

joostlek commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -109,64 +128,35 @@ def __init__(
+            try:
+                await self.api.login()
+            except InvalidAuth as e:
+                raise PlatformNotReady(
+                    translation_domain=DOMAIN,
+                    translation_key="authentication_exception",
+                    translation_placeholders={CONF_USERNAME: self.api.username},
+                ) from e
```
Let's not do translations at this point

joostlek commented on tests/components/pyload/conftest.py:
```diff
@@ -0,0 +1,74 @@
+
+
+@pytest.fixture(name="pyload_config")
+def mock_pyload_config_entry() -> ConfigType:
```
config entry?

joostlek commented on tests/components/pyload/test_sensor.py:
```diff
@@ -0,0 +1,109 @@
+
+    mock_pyloadapi.get_status.side_effect = exception
+
+    new_sensor = PyLoadSensor(
+        api=mock_pyloadapi,
+        sensor_type=SENSOR_TYPES["speed"],
+        client_name=pyload_config[DOMAIN][CONF_NAME],
+    )
```
Let's not create sensors here, we should test them via the state machine

tr4nt0r replied:
If there were a config_entry i would test if it is in state SETUP_RETRY or SETUP_ERROR, but not sure how to do it here. Nevertheless, in the following PRs the setup in configuration.yaml will be deprecated anyway so not going to put too much effort in it

joostlek commented on tests/components/pyload/test_sensor.py:
```diff
@@ -0,0 +1,109 @@
+    mock_pyloadapi.get_status.side_effect = InvalidAuth
+    mock_pyloadapi.login.side_effect = InvalidAuth
+
+    new_sensor = PyLoadSensor(
+        api=mock_pyloadapi,
+        sensor_type=SENSOR_TYPES["speed"],
+        client_name=pyload_config[DOMAIN][CONF_NAME],
+    )
```
idem

tr4nt0r replied:
removed, think this is redundant

joostlek commented on tests/components/pyload/test_sensor.py:
```diff
@@ -0,0 +1,109 @@
+    mock_pyloadapi.login.side_effect = exception
+
+    with pytest.raises(expected_exception):
+        await async_setup_platform(hass, pyload_config[DOMAIN], mock_async_add_entities)
```
Let's also not call `async_setup_platform` from here

Commit:
Improve logging and exception handling

- Modify manifest.json to update logger configuration.
- Improve error messages for authentication failures in sensor.py.
- Simplify and rename pytest fixtures in conftest.py.
- Update test cases in test_sensor.py to check for log entries and remove unnecessary code.

tr4nt0r commented on homeassistant/components/pyload/sensor.py:
```diff
@@ -77,16 +80,32 @@ def setup_platform(
-    url = f"{protocol}://{host}:{port}/api/"
+    url = f"{protocol}://{host}:{port}/"
 
+    session = async_create_clientsession(
+        hass,
+        verify_ssl=False,
+        cookie_jar=CookieJar(unsafe=True),
```
This is required to be able to connect with an ip address instead of a hostname. The service is usually only accessible from the local network

tr4nt0r replied:
This is secure enough for a local service, no change required.

tr4nt0r commented:
@coderabbitai resolve

Commit:
remove exception translations
